src/pages/ShareCodePage.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import {View, ScrollView} from 'react-native';
+import ScreenWrapper from '../components/ScreenWrapper';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import Navigation from '../libs/Navigation/Navigation';
+import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
+import QRShareWithDownload from '../components/QRShare/QRShareWithDownload';
+import compose from '../libs/compose';
+import reportPropTypes from './reportPropTypes';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../components/withCurrentUserPersonalDetails';
+import styles from '../styles/styles';
+import roomAvatar from '../../assets/images/avatars/room.png';
+import * as ReportUtils from '../libs/ReportUtils';
+import MenuItem from '../components/MenuItem';
+import Clipboard from '../libs/Clipboard';
+import * as Expensicons from '../components/Icon/Expensicons';
+import getPlatform from '../libs/getPlatform';
+import CONST from '../CONST';
+
+const propTypes = {
+    /** The report currently being looked at */
+    report: reportPropTypes,
+
+    ...withLocalizePropTypes,
+    ...withCurrentUserPersonalDetailsPropTypes,
+};
+
+const defaultProps = {
+    report: undefined,
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+// eslint-disable-next-line react/prefer-stateless-function
+class ShareCodePage extends React.Component {
+    qrCodeRef = React.createRef();
+
+    render() {
+        const isReport = this.props.report != null && this.props.report.reportID != null;
+
+        const url = isReport ? `${CONST.NEW_EXPENSIFY_URL}r/${this.props.report.reportID}` : `${CONST.NEW_EXPENSIFY_URL}details?login=${this.props.session.email}`;
+
+        const platform = getPlatform();
+        const isNative = platform === CONST.PLATFORM.IOS || platform === CONST.PLATFORM.ANDROID;
```

Comment:
@chrispader This is against our guide on how to do platform code. Please refer to https://github.com/Expensify/App#platform-specific-file-extensions and submit a PR that refactors this appropriately.

Reply:
This is actually just a temporary solution until QR code downloads are also available on web. We're currently working on that [here](https://github.com/Expensify/App/issues/19834).

Once this is possible, the platform-specific code will be removed.

Reply:
Yep, we unfortunately had to push this out in time for EC3- but, we're cleaning it up as soon as possible once we figure out the CORs issue.